### DISCUSSION TITLE
how to insert Enum value from record?

### DISCUSSION THREAD

**Discussion by Irvingouj** [pocketbase/pocketbase#5244] [Q&A] [answered]
Here's my code 
```go
func fetchAndSavePrices(app *pocketbase.PocketBase) error {
	params := &stripe.PriceSearchParams{
		SearchParams: stripe.SearchParams{
			Query: "active:'true'",
		},
	}
	result := price.Search(params)
	priceCollection, err := app.Dao().FindCollectionByNameOrId("price")
	if err != nil {
		app.Logger().Error("Error fetching price collection", "error", err)
		return err
	}

	for _, price := range result.PriceSearchResult().Data {
		record := models.NewRecord(priceCollection)
		record.Set("lookup_key", price.LookupKey)
		record.Set("unit_amount", price.UnitAmount)
		record.Set("currency", price.Currency)
		record.Set("price_id", price.ID)

		if err := app.Dao().Save(record); err != nil {
			app.Logger().Error("Error saving price", "error", err)
			return err
		}
	}
	return nil
}
```

and my table looks like this
![image](https://github.com/user-attachments/assets/e14ee56d-b475-470c-83ff-0f9332b2639d)

I put debugger at `record.Set("currency", price.Currency)` and the value of `price.Currency` is 
```
price.Currency
"usd"
```
the currency colunm is N/A always
![image](https://github.com/user-attachments/assets/8d64e7c1-49bf-4e87-a134-e53fda7aee50)

The problem is that I could not insert the enum value even if it matches one of the string value exactly.

**Comment by ganigeorgiev** [comment, marked as answer]
You need to call `app.Dao().SaveRecord`, not `app.Dao().Save` (this is simplified with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and there will be only `app.Save`/`app.SaveNoValidate`) although in your case if your collection is `base` it shouldn't matter. If you are still not able to resolve it, please try to reproduce it with a more simplified version without the Stripe related structs. Also make sure that there is no `OnModelBeforeCreate` hook that resets the value.

_Note that `app.Dao().SaveRecord()` doesn't perform validations. If you want validations currently you have to use [forms.RecordUpsert](https://pocketbase.io/docs/go-records/#create-new-record-with-data-validations) (again this is simplified with the refactoring and the form will be no longer needed)._

**Reply by Irvingouj** [reply]
Thank you very much, problem found, the stripe currency type is an alias to string.
